job-server/src/spark.jobserver/JobManagerActor.scala

```diff
@@ -230,10 +224,10 @@ class JobManagerActor(dao: JobDAO,
       // NOTE: This may not even be necessary if we set the driver ActorSystem classloader correctly
       Thread.currentThread.setContextClassLoader(jarLoader)
       val job = constructor()
-      if (job.isInstanceOf[NamedRddSupport]) {
-        val namedRdds = job.asInstanceOf[NamedRddSupport].namedRddsPrivate
-        if (namedRdds.get() == null) {
-          namedRdds.compareAndSet(null, new JobServerNamedRdds(rddManagerActor))
+      if (job.isInstanceOf[NamedObjectSupport]) {
+        val namedObjects = job.asInstanceOf[NamedObjectSupport].namedObjectsPrivate
+        if (namedObjects.get() == null) {
+          namedObjects.compareAndSet(null, new JobServerNamedObjects(context.system))
```

Comment:
I'd like to see if we could do a slightly less hacky version of this.  If everyone uses NamedObjectSupport then no need to mix it in.  Maybe put the logic directly in ContextLike.